frontend/src/components/TableSkeleton/index.vue

```diff
@@ -0,0 +1,40 @@
+<script lang="ts" setup>
+import Skeleton from 'primevue/skeleton';
+import DataTable from 'primevue/datatable';
+import Column from 'primevue/column';
+
+import type { ColumnType } from '@/types/TableTypes';
+
+const props = defineProps<{
+    headers: string[],
+    rowAmount: number,
+    className?: string,
+}>();
+
+// Create the columns array using headers
+const columns: ColumnType[] = props.headers.map((header) => ({
+    field: header,
+    header: header,
+}));
+
+// Create placeholder rows for the skeleton table based on rowAmount
+const skeletonRows = Array.from({ length: props.rowAmount }).map(() =>
+    // Create a placeholder object with empty values for each column
+    columns.reduce((acc, col) => {
+        acc[col.field] = ''; // Set empty value for each column's field
+        return acc;
+    }, {} as Record<string, string>)
+);
```

Comment:
It was hard for me to write too 🤣

The skeletonRows logic is used to create placeholder rows that match the structure expected by the DataTable. Your suggestion of using `Array.from({ length: props.rowAmount })` can simplify generating the array, but it won’t provide the right structure by itself, it would just create an array of undefined values.

The reduce part in the current code ensures each row has the correct fields to match the columns in the DataTable. To make it simpler while still keeping the necessary structure, we could do something like 
```javascript
Array.from({ length: props.rowAmount }, () => 
  Object.fromEntries(columns.map(col => [col.field, '']))
)
```

so it's easier to understand
done here https://github.com/bcgov/nr-forests-access-management/pull/1607/commits/b61932dee35eef0d2faa83604fc792d48ea54dce

Reply:
I am not sure if it matters for the correct structure for this skeleton DataTable, since it take "any[]" type from the `:value=`, when I tried it it does not seem to have problem, but maybe I missed something to see.

But it is clear a bit now. Thanks!